File: java/resistor-color-duo/src/main/java/ResistorColorDuo.java
import static java.util.Map.entry;

import java.util.Map;

class ResistorColorDuo {
  static final Map<String, Integer> COLOR_TO_DIGIT =
      Map.ofEntries(
          entry("black", 0),
          entry("brown", 1),
          entry("red", 2),
          entry("orange", 3),
          entry("yellow", 4),
          entry("green", 5),
          entry("blue", 6),
          entry("violet", 7),
          entry("grey", 8),
          entry("white", 9));

  int value(String[] colors) {
    return COLOR_TO_DIGIT.get(colors[0]) * 10 + COLOR_TO_DIGIT.get(colors[1]);
  }
}

File: python/resistor-color-duo/resistor_color_duo.py
COLOR_TO_DIGIT = {
    'black': 0,
    'brown': 1,
    'red': 2,
    'orange': 3,
    'yellow': 4,
    'green': 5,
    'blue': 6,
    'violet': 7,
    'grey': 8,
    'white': 9
}


def value(colors):
    return COLOR_TO_DIGIT[colors[0]] * 10 + COLOR_TO_DIGIT[colors[1]]
